src/test/java/org/prebid/server/bidder/videobyte/VideobyteBidderTest.java

```diff
@@ -0,0 +1,286 @@
+import com.iab.openrtb.response.SeatBid;
+import io.vertx.core.MultiMap;
+import org.junit.Before;
+import org.junit.Test;
+import org.prebid.server.VertxTest;
+import org.prebid.server.bidder.model.BidderBid;
+import org.prebid.server.bidder.model.BidderError;
+import org.prebid.server.bidder.model.HttpCall;
+import org.prebid.server.bidder.model.HttpRequest;
+import org.prebid.server.bidder.model.HttpResponse;
+import org.prebid.server.bidder.model.Result;
+import org.prebid.server.proto.openrtb.ext.ExtPrebid;
+import org.prebid.server.proto.openrtb.ext.request.videobyte.ExtImpVideobyte;
+import org.prebid.server.proto.openrtb.ext.response.BidType;
+import org.prebid.server.util.HttpUtil;
+
+import java.util.Arrays;
+import java.util.Collections;
+import java.util.List;
+import java.util.Map;
+import java.util.function.UnaryOperator;
+import java.util.stream.Collectors;
+
+import static java.util.function.UnaryOperator.identity;
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.assertj.core.api.Assertions.assertThatIllegalArgumentException;
+import static org.assertj.core.api.Assertions.tuple;
+
+public class VideobyteBidderTest extends VertxTest {
+
+    private static final String ENDPOINT_URL = "https://site.st/uri";
+
+    private VideobyteBidder bidder;
+
+    @Before
+    public void setUp() {
+        bidder = new VideobyteBidder(ENDPOINT_URL, jacksonMapper);
+    }
+
+    @Test
+    public void creationShouldFailOnInvalidEndpointUrl() {
+        assertThatIllegalArgumentException().isThrownBy(() -> new VideobyteBidder("invalid url", jacksonMapper));
+    }
+
+    @Test
+    public void makeHttpRequestsShouldReturnErrorWhenImpExtCouldNotBeParsed() {
+        // given
+        final BidRequest bidRequest = givenBidRequest(givenImp(identity()),
+                givenImp(imp -> imp.ext(mapper.valueToTree(ExtPrebid.of(null, mapper.createArrayNode())))));
+
+        // when
+        final Result<List<HttpRequest<BidRequest>>> result = bidder.makeHttpRequests(bidRequest);
+
+        // then
+        assertThat(result.getValue()).hasSize(1);
+        assertThat(result.getErrors()).hasSize(1)
+                .extracting(BidderError::getMessage)
+                .allMatch(errorMessage -> errorMessage.startsWith("Ignoring imp id=null, error while decoding, err:"));
+    }
+
+    @Test
+    public void makeHttpRequestsShouldCreateRequestForEachImp() {
+        // given
+        final BidRequest bidRequest = givenBidRequest(givenImp(identity()), givenImp(identity()));
+
+        // when
+        final Result<List<HttpRequest<BidRequest>>> result = bidder.makeHttpRequests(bidRequest);
+
+        // then
+        assertThat(result.getValue()).hasSize(2)
+                .extracting(HttpRequest::getPayload)
+                .extracting(BidRequest::getImp)
+                .extracting(List::size)
+                .containsOnly(1);
+        assertThat(result.getErrors()).isEmpty();
```

Comment:
Pls, check exact results, not only their sizes.

Reply:
This test checks that for each imp will be created new request. Not for content.

Reply:
I see, but we can test both size and content at once.

Reply:
For now, the Vidobyte bidder isn't modifying the impression. So I see no reason why we need to test content. If in the future the bidder modifies the impressions, the necessary tests will be written.